plugins/com.aptana.editor.common/src/com/aptana/editor/common/CommonUtil.java

```diff
@@ -31,4 +48,46 @@ public static IToken getToken(String tokenName)
 		return new Token(tokenName);
 	}
 
+	/**
+	 * Registering double click listener to the project explorer viewer to handle
+	 * "Open no extension files with JavaScript Source Editor" preference option. Which is added under the "Editors"
+	 * section. Based on the preference option while opening a file will overwrite the default editor id of a file.
+	 */
+	public static void handleOpenWithEditorPref()
+	{
+		if (!isDoubleClickRegistered)
```

Comment:
I hope this method won't be called by multiple threads at the same time, which will result in adding multiple doubleClick listeners. In order to avoid that, we can make the method synchronized.